ts_src/psetv2/input.ts

```diff
@@ -610,30 +610,52 @@ export class PsetInput {
       );
     }
     const issuanceValueCommitSet =
-      this.issuanceValueCommitment! && this.issuanceValueCommitment.length > 0;
+      this.issuanceValueCommitment && this.issuanceValueCommitment.length > 0;
     const issuanceBlindValueProofSet =
-      this.issuanceBlindValueProof! && this.issuanceBlindValueProof.length > 0;
+      this.issuanceBlindValueProof && this.issuanceBlindValueProof.length > 0;
     if (
-      this.issuanceValue! &&
+      this.issuanceValue &&
       issuanceValueCommitSet !== issuanceBlindValueProofSet
     ) {
       throw new Error('Missing input issuance value commitment or blind proof');
     }
     const issuanceInflationKeysCommitSet =
-      this.issuanceInflationKeysCommitment! &&
+      this.issuanceInflationKeysCommitment &&
       this.issuanceInflationKeysCommitment.length > 0;
     const issuanceBlindInflationKeysProofSet =
-      this.issuanceBlindInflationKeysProof! &&
+      this.issuanceBlindInflationKeysProof &&
       this.issuanceBlindInflationKeysProof.length > 0;
     if (
-      this.issuanceInflationKeys! &&
+      this.issuanceInflationKeys &&
       issuanceInflationKeysCommitSet !== issuanceBlindInflationKeysProofSet
     ) {
       throw new Error(
         'Missing input issuance inflation keys commitment or blind proof',
       );
     }
 
+    // issuance case
+    if (
+      this.issuanceBlindingNonce &&
+      this.issuanceValue &&
+      this.issuanceInflationKeys
+    ) {
+      if (this.issuanceBlindingNonce.equals(ZERO)) {
+        if (this.issuanceValue <= 0 && this.issuanceInflationKeys <= 0) {
```

Comment:
this condition won't ever be true since just some line above we want value and token to be defined AND > 0.

In general, to make sure the input issuance is well constructed we should make sure that:
* if issuance or token amounts are defined, the blinding nonce is not null
* in case of issuance, issuance or token values are > 0
* in case of reissuance, the issuance value is > 0 and token value == 0

Reply:
second and third checks make the sanity check too much strict. Let's just make sure that issuance value|token and blinding nonce are defined